web3/method.py

```diff
@@ -166,28 +157,28 @@ def __get__(
             )
 
         provider = module.w3.provider
-        if hasattr(provider, "_is_batching") and provider._is_batching:
+        if getattr(provider, "_is_batching", False):
             if self.json_rpc_method in RPC_METHODS_UNSUPPORTED_DURING_BATCH:
                 raise MethodNotSupported(
                     f"Method `{self.json_rpc_method}` is not supported within a batch "
                     "request."
                 )
-            return module.retrieve_request_information(self)
+            return module.retrieve_request_information(self)  # type: ignore [return-value]
```

Comment:
not really sure how to deal with the line len err here, but everything else is ready for review